### PR TITLE
Fix custom bounds handling in test problems

### PR DESCRIPTION
**Pull request by saitcakmak**
Summary: Fixes https://github.com/pytorch/botorch/issues/1759

Differential Revision: D44326798

### CODE REVIEW

File: test/test_functions/test_synthetic.py [hunk 1]
```diff
@@ -45,8 +47,34 @@ class DummySyntheticTestFunctionWithOptimizers(DummySyntheticTestFunction):
 
 
 class TestSyntheticTestFunction(BotorchTestCase):
+    functions_with_custom_bounds = [
```
Comment (Balandat) [inline test/test_functions/test_synthetic.py:50]: maybe add a comment what the int value here is

**Review by Balandat — COMMENTED** [review]
Some tests are still failing, o/w this generally lgtm, thanks.

File: test/test_functions/test_synthetic.py [hunk 2]
```diff
@@ -57,6 +85,19 @@ def test_custom_bounds(self):
...
+            )
+        )
+
+        # Test each function with custom bounds.
+        for func_class, dim in self.functions_with_custom_bounds:
+            bounds = [(-1e5, 1e5) for _ in range(dim)]
+            bounds_tensor = torch.tensor(bounds).T
+            func = func_class(bounds=bounds)
+            self.assertEqual(func._bounds, bounds)
+            self.assertTrue(torch.allclose(func.bounds, bounds_tensor))
```
Comment (esantorella) [inline test/test_functions/test_synthetic.py:100]: Thank you for this!

**Comment by saitcakmak** [comment]
> Do these problems need to have a _bounds attribute at all? It seems like self._bounds is never accessed after it’s set. ([code search](https://github.com/search?q=repo%3Apytorch%2Fbotorch+self._bounds&type=code)) I feel a lot better about having both now that we have the unit test that they contain the same values though!

Do they need to? Probably not. They've had it for a long time, so it makes sense to keep it around. I generally wouldn't rely on code search to find usage of test problems since they'll mostly be used for benchmarking by us or external users. I guess this makes it a bit harder to catch non-breaking bugs with them as well.

**Pull request closed without merging**